### PR TITLE
[Network]`az network vnet-gateway`: `--vpn-auth-type` allow multi value

### PR DESCRIPTION
**Pull request by msyyc**
**Description**

https://github.com/Azure/azure-cli/issues/17260
https://github.com/Azure/azure-cli/issues/17482
**Testing Guide**

test_network_vnet_gateway_multi_auth
**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -6371,6 +6372,16 @@ def create_vnet_gateway(cmd, resource_group_name, virtual_network_gateway_name,
...
 
+        # multi authentication
+        if cmd.supported_api_version(min_api='2020-11-01'):
+            vnet_gateway.vpn_client_configuration.vpn_authentication_types = vpn_auth_type
+            vnet_gateway.vpn_client_configuration.aad_tenant = aad_tenant
+            vnet_gateway.vpn_client_configuration.aad_issuer = aad_issuer
+            vnet_gateway.vpn_client_configuration.aad_audience = aad_audience
+            vnet_gateway.vpn_client_configuration.vpn_client_root_certificates = [
+                VpnClientRootCertificate(name=root_cert_name,
+                                         public_cert_data=root_cert_data)] if root_cert_data else None
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:6383]: Do we need to throw error if user provide rout_cert_data but forget root_cert_name?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -6401,6 +6414,15 @@ def update_vnet_gateway(cmd, instance, sku=None, vpn_type=None, tags=None,
...
         c.set_param('radius_server_secret', radius_secret)
+        if cmd.supported_api_version(min_api='2020-11-01'):
+            c.set_param('aad_tenant', aad_tenant)
+            c.set_param('aad_audience', aad_audience)
+            c.set_param('aad_issuer', aad_issuer)
+            c.set_param('vpn_authentication_types', vpn_auth_type)
+
+    if root_cert_data and cmd.supported_api_version(min_api='2020-11-01'):
+        upsert_to_collection(instance.vpn_client_configuration, 'vpn_client_root_certificates',
+                             VpnClientRootCertificate(name=root_cert_name, public_cert_data=root_cert_data), 'name')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:6425]: Same question as above

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 1 above)
Comment (msyyc) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:6383]: root_cert_data is necessary while root_cert_name is not.

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 2 above)
Comment (msyyc) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:6425]: Same reply as above

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**